contracts/DepegProduct.sol

```diff
@@ -138,26 +128,38 @@ contract DepegProduct is
         uint256 riskpoolId
     )
         Product(productName, token, POLICY_FLOW, riskpoolId, registry)
-        EIP712(EIP712_DOMAIN_NAME, EIP712_DOMAIN_VERSION)
     {
         // initial product state is active
         _state = DepegState.Active;
 
         require(priceDataProvider != address(0), "ERROR:DP-001:PROVIDER_ZERO");
         _priceDataProvider = IPriceDataProvider(priceDataProvider);
-
-        _tokenContract = IERC20Metadata(token);
-        _protectedToken = _priceDataProvider.getToken();
-        require(_protectedToken != token, "ERROR:DP-002:SAME_TOKEN");
+        require(token != _priceDataProvider.getToken(), "ERROR:DP-002:SAME_TOKEN");
 
         IComponent poolComponent = _instanceService.getComponent(riskpoolId); 
         address poolAddress = address(poolComponent);
 
         _riskpool = DepegRiskpool(poolAddress);
         _treasury = TreasuryModule(_instanceService.getTreasuryAddress());
-        _depeggedBlockNumber = 0;
     }
 
+
+    function setMessageHelper(address depegMessageHelper)
```

Comment:
wouldn't it be better to set this in the constructor as its **not optional** because without the message helper the `applyForPolicyWithBundleAndSignature` wouldn't work. Or alternatively check that the `messageHelper` in a modifier of said method.

Reply:
moved to constructor